lib/utapi/utilities.js

```diff
@@ -0,0 +1,117 @@
+                            timeRange,
+                            accessKey,
+                            secretKey,
+                            verbose,
+                            ssl) {
+    const options = {
+        host,
+        port,
+        method: 'POST',
+        path: '/buckets?Action=ListMetrics',
+        headers: {
+            'content-type': 'application/json',
+            'cache-control': 'no-cache',
+        },
+        rejectUnauthorized: false,
+    };
+    const transport = ssl ? https : http;
+    const request = transport.request(options, response => {
+        if (verbose) {
+            logger.info('response status code', {
+                statusCode: response.statusCode,
+            });
+            logger.info('response headers', { headers: response.headers });
+        }
+        const body = [];
+        response.setEncoding('utf8');
+        response.on('data', chunk => body.push(chunk));
+        response.on('end', () => {
+            const responseBody = JSON.parse(body.join(''));
+            if (response.statusCode >= 200 && response.statusCode < 300) {
+                // eslint-disable-next-line no-console
+                console.log(responseBody);
+                process.exit(0);
+            } else {
+                logger.error('request failed with HTTP Status ', {
+                    statusCode: response.statusCode,
+                    body: responseBody,
+                });
+                process.exit(1);
+            }
+        });
+    });
+    auth.generateV4Headers(request, '', accessKey, secretKey, 's3');
+    if (verbose) {
+        logger.info('request headers', { headers: request._headers });
+    }
+    request.write(JSON.stringify({ buckets, timeRange }));
+    request.end();
+}
+
+/**
+ * This function is used as a binary to send a request to utapi server
+ * to list bucket metrics
+ *
+ * @return {undefined}
+ */
+export function listBucketMetrics() {
+    commander
+        .version('0.0.1')
+        .option('-a, --access-key <accessKey>', 'Access key id')
+        .option('-k, --secret-key <secretKey>', 'Secret access key')
+        .option('-b, --buckets <buckets>', 'Name of bucket(s)' +
+        'with a comma separator if more than one')
+        .option('-s, --start <start>', 'Start of time range')
+        .option('-e --end <end>', 'End of time range')
+        .option('-h, --host <host>', 'Host of the server')
+        .option('-p, --port <port>', 'Port of the server')
+        .option('--ssl', 'Enable ssl')
+        .option('-v, --verbose')
+        .parse(process.argv);
+
+    const { host, port, accessKey, secretKey, start, end,
+        buckets, verbose, ssl } =
+        commander;
+    const requiredOptions = { host, port, accessKey, secretKey, buckets,
+        start };
+    Object.keys(requiredOptions).forEach(option => {
+        if (!requiredOptions[option]) {
+            logger.error(`missing required option: ${option}`);
```

Comment:
I think it's mostly about what it's used for, and how we want to show the output. For instance, if it's something automated, it's fine to have parseable output (json would be a good choice then). If not, maybe we'd lean towards a more classical approach. Not an issue anyways, just wanted to know if there was a choice there or not.